src/rust/ide/lib/span-tree/src/node.rs

```diff
@@ -172,6 +373,7 @@ impl<'a> Ref<'a> {
         enso_data::text::Span::new(self.span_begin,self.node.size)
     }
 
+    // FIXME: These docs are invalid, this function is deeply magical
     /// Get the reference to child with given index. Returns None if index if out of bounds.
     pub fn child(mut self, index:usize) -> FallibleResult<Ref<'a>> {
```

Comment:
I don't understand. Have you test which shows that docs are invalid?

The only magic thing is that we return our `Ref` structure instead of plain reference, but Ref structure have more useful information in it (like the crumbs to the current node).

Reply:
Ok, so the function is not as magical as I tohught, but also it doesnt return none. Im fixing docs.